src/pages/student/settings/page.tsx

```diff
@@ -1,11 +1,186 @@
-import { Layout } from "antd";
 import { PageHeader } from "../../../shared/ui/PageHeader/PageHeader";
+import { Button, Form, Input, Layout } from "antd";
+import { useUnit } from "effector-react";
+import {
+  toStudentPage,
+  setTelegramNickName,
+  setGithubNickName,
+  setRepositoryReact,
+  $telegramNickName,
+  $githubNickName,
+  $repositoryHtml,
+  $repositoryReact,
+  setRepositoryHtml,
+} from "../model";
+import {
+  repository,
+  Repository,
+} from './mocks';
+
+import "./page.css";
 
 const { Content } = Layout;
 
-export const StudentSettingsPage = () => (
+ 
+
+export const StudentSettingsPage = () => {
+  const [
+    telegramNickName,
+    githubNickName,
+    repositoryHtml,
+    repositoryReact,
+    toStudentPageFn,
+    setTelegramNickNameFn,
+    setGithubNickNameFn,
+    setRepositoryHtmlFn,
+    setRepositoryReactFn
+  ] = useUnit([
+    $telegramNickName,
+    $githubNickName,
+    $repositoryHtml,
+    $repositoryReact,
+    toStudentPage,
+    setTelegramNickName,
+    setGithubNickName,
+    setRepositoryHtml,
+    setRepositoryReact
+  ])
+  const handleFinish = () => toStudentPageFn();
+  const repositories = repository.map( obj => {
+    return (
+      <Form.Item
+
+        key={obj.description}
+        name={obj.name}
+        rules={[
+          {
+            required: true,
+            message: 'Please input name of repository!'
+          },
+          {
+            pattern: /[0-9a-z_]*$/,
+            message: "Please input name of repository!",
+          }
+        ]}
+        >
+    <Input
+      placeholder="имя репозитория с версткой"
+      value={obj.name}
+      onChange={({ target: { value } }) => setRepositoryFn(value)}
+      allowClear
+    />
+    </Form.Item>
+    
+    )}
+    );
+
+
+  return(
   <Layout>
     <PageHeader title="Настройки" />
-    <Content />
+    <Content className="Content__settings">
+      <Form
+        name="userLinks"
+        className="Form"
+        wrapperCol={{ span: 4, offset: 1 }}
+        onFinish={handleFinish}
+        autoComplete="off"
+        validateTrigger="onSubmit"
+      >
+        <Form.Item
+        fieldId="github"
+          name="githubNickName"
+          rules={[
+            { 
+              required: true,
+              message: 'Please input your github nickname!'
+            },
+            {
+              pattern: /[0-9a-z_]*$/,
+              message: "githubNickName",
+            }
+          ]}
+        >
+          <Input
+            placeholder="ник в гитхабе"
+            value={githubNickName}
+            onChange={ ({ target: { value } }) => setGithubNickNameFn(value) }
+            allowClear
+          />
+        </Form.Item>
+        <Form.Item
+          fieldId=""
+          name="telegramNickName"
+          rules={[
+            {
+              required: true,
+              message: 'Please input your telegram nickname!'
+            },
+            {
+              pattern: /[0-9A-Za-z_]*$/,
```

Comment:
If we suppose to get tg nickname started from @ , pattern should be like this /^@[0-9A-Za-z_]*$/